src/commands/deploy.ts

```diff
@@ -102,16 +118,10 @@ export async function deploy(context: IActionContext, confirmDeployment: boolean
 
     const correlationId = getRandomHexString();
     context.telemetry.properties.correlationId = correlationId;
-    const siteConfig: WebSiteModels.SiteConfigResource = await node.root.client.getSiteConfig();
 
-    if (!fsPath) {
-        if (javaUtils.isJavaRuntime(siteConfig.linuxFxVersion)) {
-            const fileExtension: string = javaUtils.getArtifactTypeByJavaRuntime(siteConfig.linuxFxVersion);
-            fsPath = await javaUtils.showQuickPickByFileExtension(context, `Select the ${fileExtension} file to deploy...`, fileExtension);
-            await javaUtils.configureJavaSEAppSettings(node);
-        } else {
-            fsPath = await workspaceUtil.showWorkspaceFoldersQuickPick("Select the folder to deploy", context, constants.configurationSettings.deploySubpath);
-        }
+    siteConfig = siteConfig ? siteConfig : await node.root.client.getSiteConfig();
+    if (javaUtils.isJavaRuntime(siteConfig.linuxFxVersion)) {
+        await javaUtils.configureJavaSEAppSettings(node);
```

Comment:
This will cause another prompt to come up after creation.  I was thinking I could get the port value from the user before creation and then just retroactively run the command after it's done (since this relies on the app being created before it can update the app settings).

Reply:
I'm fine with prompting for port after creation in the Java case.